front/src/HOC/withLogger.js

```diff
@@ -0,0 +1,67 @@
+import React, { Component } from 'react';
+
+const getDisplayName = SomeComponent => {
+  return (
+    SomeComponent.displayName || SomeComponent.name || 'Component'
+  );
+};
+
+export const withLogger = (params = {}) => {
+  const { output, lifecycles = [] } = params;
+  const logProps = lifecycles.findIndex(el => el === 'props') !== -1;
+  const logDidMount =
+    lifecycles.findIndex(el => el === 'didMount') !== -1;
+  const logShouldUpdate =
+    lifecycles.findIndex(el => el === 'shouldUpdate') !== -1;
+  const logDidUpdate =
+    lifecycles.findIndex(el => el === 'didUpdate') !== -1;
+  const logWillUnmount =
+    lifecycles.findIndex(el => el === 'willUnmount') !== -1;
+  const log = data => {
+    if (output) {
+      return output(data);
+    }
+    console.log(data);
+  };
+
+  return ComponentToWrap => {
+    const WrappedComponent = class MyClass extends Component {
```

Comment:
Can we somehow conditionally override those lifecycle methods to avoid calling them unless they necessary?

Reply:
I think we can prepare different functions for each lifecycle and put them there.
Not sure if we can do something like:
 shouldComponentUpdate = () => {}
in component.